packages/backend/src/models/entities/Event.ts

```diff
@@ -0,0 +1,79 @@
+import { Entity, Index, Column, PrimaryColumn, OneToOne, JoinColumn } from 'typeorm';
+import { id } from '../id.js';
+import { noteVisibilities } from '../../types.js';
+import { Note } from './Note.js';
+import type { User } from './User.js';
+
+@Entity()
+export class Event {
+	@PrimaryColumn(id())
+	public noteId: Note['id'];
+
+	@OneToOne(type => Note, {
+		onDelete: 'CASCADE',
+	})
+	@JoinColumn()
+	public note: Note | null;
+
+	@Index()
+	@Column('timestamp with time zone', {
+		comment: 'The start time of the event',
+	})
+	public start: Date;
+
+	@Column('timestamp with time zone', {
+		comment: 'The end of the event',
+		nullable: true,
+	})
+	public end: Date;
+
+	@Column({
+		type: 'varchar',
+		length: 128,
+		comment: 'short name of event',
+	})
+	public title: string;
+
+	@Column('jsonb', {
+		default: {},
+		comment: 'metadata mapping for event with more user configurable optional information',
+	})
+	public metadata: Record<string, string>;
```

Comment:
Would it be better to have this be an arbitrary object type to accept more customized event schema? For example an address field (https://schema.org/PostalAddress). The drawback with this is displaying/searching a more complicated object would be more difficult than just text. A GraphQL API might be better suited for this, but we don't have one.

Reply:
It would be better to fix it to a schema like PostalAddress. Accepting arbitrary fields would probably result in too many notational gaps and would be practically unusable. Instead, it is more user-friendly for the software to indicate the fields that should be set.